### DISCUSSION TITLE
Can the go-pg specify different database drivers for different types of pg databases?

### DISCUSSION THREAD

**Discussion by UESTC-zzy** [go-pg/pg#2010] [Q&A]
Can the go-pg specify different database drivers for different types of pg databases?

**Comment by elliotcourant** [comment]
`go-pg` can connect to any database that is compatible with the PostgreSQL wire protocol as documented here: https://www.postgresql.org/docs/current/protocol.html

This includes databases like Greenplum, TimescaleDB, YugabyteDB and CockroachDB.

For example, `go-pg` is listed as compatible with Cockroach as a "Data Access Framework": https://www.cockroachlabs.com/docs/v21.1/third-party-database-tools.html#data-access-frameworks-e-g-orms

**Reply by UESTC-zzy** [reply]
OK, I will compare whether the database meets the protocol mentioned by you. Thank you very much. Another question is whether it is possible to provide a document about go-pg implementing CopyTo and CopyFrom. We would like to refer to this document.
